sessiontxn/staleread/provider.go

```diff
@@ -90,6 +91,11 @@ func (p *StalenessTxnContextProvider) OnStmtStart(_ context.Context) error {
 	return nil
 }
 
+// ActivateTxn activates the transaction.
+func (p *StalenessTxnContextProvider) ActivateTxn() (kv.Transaction, error) {
+	return nil, nil
```

Comment:
I think we should also support `ActivateTxn` for `StalenessTxnContextProvider`

Reply:
Ok, I have added some code for this.